kornia/image/base.py

```diff
@@ -21,3 +23,47 @@ class ImageSize:
     """
     height: int | Tensor
     width: int | Tensor
+
+
+class PixelFormat(Enum):
+    r"""Enum that represents the pixel format of an image."""
+    GRAY = 0
+    RGB = 1
+    BGR = 2
+
+
+class ChannelsOrder(Enum):
+    r"""Enum that represents the channels order of an image."""
+    CHANNELS_FIRST = 0
+    CHANNELS_LAST = 1
+
+
+@dataclass(frozen=True)
+class ImageLayout:
+    """Data class to represent the layout of an image.
+
+    Args:
+        image_size: image size.
+        channels: number of channels.
+        pixel_format: pixel format.
+        channels_order: channels order.
+
+    Example:
+        >>> layout = ImageLayout(ImageSize(3, 4), 3, PixelFormat.RGB, ChannelsOrder.CHANNELS_LAST)
+        >>> layout.image_size
+        ImageSize(height=3, width=4)
+        >>> layout.channels
+        3
+        >>> layout.pixel_format
+        <PixelFormat.RGB: 1>
+        >>> layout.channels_order
+        <ChannelsOrder.CHANNELS_LAST: 1>
+    """
+
```

Comment:
perhaps we can have bit size here? as default equals 8

Reply:
ok -- i'll iterate over the `PixelFormat` structure

Reply:
@johnnv1 jsut updated a bit. I made so that it matches a bit it the new Sophus2 c++ image api /c @cjpurackal 
https://github.com/strasdat/Sophus/blob/sophus2/protos/sophus/image.proto